src/pages/signing_official_console/ManageResearcherDAAsTable.jsx

```diff
@@ -0,0 +1,275 @@
+    );
+  }, [researchers]);
+
+  useEffect(() => {
+    recalculateVisibleTable({
+      tableSize, pageCount,
+      filteredList: filteredResearchers,
+      currentPage,
+      setPageCount,
+      setCurrentPage,
+      setVisibleList: setVisibleResearchers
+    });
+  }, [tableSize, pageCount, filteredResearchers, currentPage]);
+
+  const goToPage = useCallback((value) => {
+    if (value >= 1 && value <= pageCount) {
+      setCurrentPage(value);
+    }
+  }, [pageCount]);
+
+  const changeTableSize = useCallback((value) => {
+    if (value > 0 && !isNaN(parseInt(value))) {
+      setTableSize(value);
+    }
+  }, []);
+
+  const paginationBar = (
+    <PaginationBar
+      pageCount={pageCount}
+      currentPage={currentPage}
+      tableSize={tableSize}
+      goToPage={goToPage}
+      changeTableSize={changeTableSize}
+    />
+  );
+
+  const processResearcherRowData = (researchers = []) => {
+    return researchers.map(researcher => {
+      const {displayName, /*count = 0,*/ roles, libraryCards} = researcher;
+      const libraryCard = !isEmpty(libraryCards) ? libraryCards[0] : {};
+      const email = researcher.email || libraryCard.userEmail;
+      const id = researcher.userId || email;
+      return [
+        displayNameCell(displayName, email, id),
+        ...dacs.map(dac => DAACell(dac, researcher, signingOfficial.institutionId, daas, refreshResearchers, setResearchers))
+      ];
+    });
+  };
+
+  return (
+    <>
+      <div style={{ display: 'flex', justifyContent: 'space-between', alignItems: 'flex-end', width: '112%', marginLeft: '-6%' }}>
+        <div style={Styles.LEFT_HEADER_SECTION}>
+          <div style={{ ...Styles.HEADER_CONTAINER, marginRight: '-7%' }}>
+            <div style={{ ...Styles.SUB_HEADER,
+              marginTop: '0',
+              fontFamily: 'Montserrat',
+              fontWeight: 600,
+              fontSize: '2.8rem'}}>
+              My Institution&apos;s Researchers
+            </div>
+            <div style={Object.assign({}, Styles.MEDIUM_DESCRIPTION, {
+              fontSize: '16px',
+              maxWidth: '60%',
+            })}>
+              Issue, Update, or Deactivate for User&apos;s ability to request access to datasets, by agreeing to
+              Data Access Committee&apos;s (DAC&apos;s) Data Access Agreements (DAAs) in the table below.
+              Issuing a checkmark in a cell for a researcher denotes your approval of that researcher
+              to request data from the listed DAC, according to its linked DAA.
+            </div>
+          </div>
+        </div>
+        <SearchBar handleSearchChange={handleSearchChange} searchRef={searchRef}/>
+      </div>
+      <SimpleTable
+        isLoading={isLoading}
+        rowData={processResearcherRowData(visibleResearchers)}
+        columnHeaders={columnHeaderData}
+        styles={styles}
+        tableSize={tableSize}
+        paginationBar={paginationBar}
+      />
+    </>
+  );
+}
```

Comment:
Lint warnings on this page:
```

src/pages/signing_official_console/ManageResearcherDAAsTable.jsx
  Line 74:2:     Missing semicolon                                                                                                    @babel/semi
  Line 86:2:     Missing semicolon                                                                                                    @babel/semi
  Line 89:10:    Trailing spaces not allowed                                                                                          no-trailing-spaces
  Line 110:138:  Trailing spaces not allowed                                                                                          no-trailing-spaces
  Line 185:6:    React Hook useEffect has a missing dependency: 'columnHeaderData'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  Line 229:42:   'roles' is assigned a value but never used                                                                           no-unused-vars
```
I think the hook issue might require some debugging since adding it creates an infinite reload problem. The usual solution to this is to pull the offending value out of the use effect in some way.

I recently had to do this in the era commons conversion and one thing that helped was to extract the calculation into a separate function. See https://github.com/DataBiosphere/duos-ui/blob/9fb4c8577b9c372217d3b8a8ac76718687059e08/src/components/ERACommons.jsx#L60 for that example.